meta-ostro/classes/supported-recipes.bbclass

```diff
@@ -0,0 +1,288 @@
+# This class is meant to be used in a distro or local conf via
+# INHERIT += "supported-recipes".
+#
+# It automatically excludes unsupported recipes from the world build,
+# without having to maintain a set of REMOVE_FROM_WORLD_pn-<recipe>
+# entries in an include file.
+#
+# In addition, there is a pre-build check whether any unsupported
+# recipes are required to complete the build. This check is optional,
+# see SUPPORTED_RECIPES_CHECK.
+#
+# Build tools and native recipes are ignored by the class itself.
+# Those are considered less important (problems in them cannot be
+# exploited on the target platform) and/or they are supported
+# by the provider of these tools.
+
+# Space-separated list of filenames of files containing recipe names.
+# Content of these files are recipes names, one per line.
+# Empty lines and lines starting with a hash are ignored.
+SUPPORTED_RECIPES ??= ""
+
+# Empty skips check, "note/warn/error/fatal" increases the logging level,
+# with "fatal" aborting the build.
+SUPPORTED_RECIPES_CHECK ??= ""
+
+# Maximum number of dependency chains to print when the check fails.
+# Depending on a recipe deep in the stack can produce a lot of output,
+# this here acts as safeguard.
+SUPPORTED_RECIPES_CHECK_DEPENDENCY_LINES ??= "50"
+
+def load_supported_recipes(d):
+    import os
+    import re
+
+    class SupportedRecipe:
+        def __init__(self, pattern, filename, linenumber):
+            self.filename = filename
+            self.pattern = pattern
+            self.linenumber = linenumber
+            parts = pattern.split('@')
+            if len(parts) != 2:
+                raise RuntimeException("%s.%d: entry must have format <recipe name regex>@<collection name regex>, splitting by @ found %d parts instead: %s" % (
+                    filename,
+                    linenumber,
+                    len(parts),
+                    pattern))
+            def parse_regex(regex):
+                try:
+                    # must match entire string, hence the '$'
+                    return (re.compile(regex + '$'), regex)
```

Comment:
if user for some reason uses $ in the end of his pattern, it shouldn't be added as last symbol.